### PR TITLE
[WIP] Hardcoded rules

### PR DESCRIPTION
**Pull request by RishabhTayal**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 ## Master
 
+##### New Rule
```
Comment (ornithocoder) [inline CHANGELOG.md]: Please add new rules to the 'Enhancements' section down below instead of creating a new section.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+* Hardcoded rules.  
+  [RishabhTayal](https://github.com/RishabhTayal)
+  [#1377](https://github.com/realm/SwiftLint/issues/issue_number)
```
Comment (ornithocoder) [inline CHANGELOG.md]: Please add the issue number to the URL.

**Comment by RishabhTayal** [comment]
@ornithocoder Thanks for reviewing. The changes to Carthage directory is not intentional. I am not sure how they got changed. How can I revert back only those changes?

**Comment by marcelofabri** [comment]
> Additionally, I would recommend you to change the title to "[WIP] Hardcoded rules" to avoid code reviews while the feature isn't complete.

Or even closing this and reopening after. There's no point in adding extra work in Circle CI and Travis if this is far from being done.

**Comment by RishabhTayal** [comment]
closing pull request. Will open later.

**Comment by ornithocoder** [comment]
@RishabhTayal should you want to test the version you're working on against the same open source software that Circle CI tests against, you can run the following command:

```swift
$ ./script/oss-check --branch BRANCH --iterations 1 -v
```

It will test the BRANCH against master, show the difference in performance between the two and list the violations that the new rule introduced.

**Pull request closed without merging**